### DISCUSSION TITLE
Splitting up the query type definitions

### DISCUSSION THREAD

**Discussion by axhamre** [hayes/pothos#391] [Q&A] [answered]
Given the example code in `https://github.com/hayes/pothos/blob/main/examples/prisma/src/schema.ts`, what would be a recommended way to split up the queries (defined in `queryType`) into a structure along the lines below?

`/gql/user/user.query.ts`
`/gql/post/post.query.ts`
`/gql/posts/posts.query.ts`

**Comment by hayes** [comment, marked as answer]
You can define the query type once, with some or no fields, then use `builder.queryField` or `builder.queryFields` to add additional fields in other files

**Reply by axhamre** [reply]
Thank you!

For anyone else seeing this, I turned this:

```
builder.queryType({
  fields: (t) => ({
    user: t.prismaField({
      type: "User",
      nullable: true,
      args: {
        id: t.arg.id({ required: true }),
      },
      resolve: (query, root, args) =>
        prisma.user.findUnique({
          ...query,
          where: { id: Number.parseInt(String(args.id), 10) },
        }),
    }),
  }),
});
```

Into this:

```
builder.queryField("user", (t) =>
  t.prismaField({
    type: "User",
    nullable: true,
    args: {
      id: t.arg.id({ required: true }),
    },
    resolve: (query, root, args) =>
      prisma.user.findUnique({
        ...query,
        where: { id: Number.parseInt(String(args.id), 10) },
      }),
  })
);
```

**Reply by hayes** [reply]
Added this example (will add a lot more in the future) that has some examples of how to build a schema thats spread across multiple files: https://github.com/hayes/pothos/tree/main/examples/complex-app
